### PR TITLE
handle button add to card

### PR DESCRIPTION
**Pull request by tri-et**

### CODE REVIEW

File: src/components/ModifiersGroup.vue [hunk 1]
```diff
@@ -26,8 +27,12 @@ export default {
   data() {
     return {
       defaultModifiers: [],
+      oldModifiers: [],
     }
   },
+  computed: {
+    ...mapGetters('modifier', ['getCurrentMenuModifier']),
```
Comment (hemvanh) [inline src/components/ModifiersGroup.vue]: getCurrentMenuModifier -> getCurrentMenuModifiers

File: src/components/ModifiersGroup.vue [hunk 2]
```diff
@@ -41,12 +46,44 @@ export default {
...
   mounted() {
     let modifierDedault = _d.find(this.data, modifier => {
       return modifier.isDefault
     })
     if (modifierDedault) this.defaultModifiers = modifierDedault
   },
+  watch: {
+    defaultModifiers: {
+      immediate: true,
+      handler(newModifiers, oldModifiers) {
```
Comment (hemvanh) [inline src/components/ModifiersGroup.vue]: handler(newModifiers, oldModifiers) {....  -> handler: this.updateModifiers

pls note: update is 1 word, not upDate

File: src/components/ModifiersGroup.vue [hunk 3]
```diff
@@ -41,12 +46,44 @@ export default {
...
+      } else newFormatedModifier = [_d.pick(this.defaultModifiers, ['id', 'price'])]
+
+      this.setCurrentMenuModifier(_d.concat(currentModifiers, newFormatedModifier))
+    },
+  },
   mounted() {
     let modifierDedault = _d.find(this.data, modifier => {
       return modifier.isDefault
     })
     if (modifierDedault) this.defaultModifiers = modifierDedault
```
Comment (hemvanh) [inline src/components/ModifiersGroup.vue]: this.defaultModifiers =_d.find(this.data, modifier => {
             return modifier.isDefault
  })

File: src/pages/MenuDetail.vue [hunk 4]
```diff
@@ -47,26 +51,44 @@ export default {
...
-    ...mapMutations('modifier', ['setRecs']),
+    ...mapMutations('modifier', ['setRecs', 'setCurrentMenuModifier']),
+    ...mapMutations('customerorder', {setRecsModifier: 'setRecs'}),
     backToMenu() {
       this.setRecs([])
+      this.setCurrentMenuModifier([])
       this.$router.go(-1)
       this.setCounter(1)
     },
+    calculatorMenuPrice() {
```
Comment (hemvanh) [inline src/pages/MenuDetail.vue]: calculatorMenuPrice -> calculateMenuPrice

File: src/pages/MenuDetail.vue [hunk 5]
```diff
@@ -47,26 +51,44 @@ export default {
...
+      this.modifierId = _.map(this.getCurrentMenuModifier, 'id').sort()
+      let menuModifierPrice =
```
Comment (hemvanh) [inline src/pages/MenuDetail.vue]: menuModifierPrice  -> menuPrice

File: src/store/customer-order/actions.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+/*
```
Comment (hemvanh) [inline src/store/customer-order/actions.js:1]: remove comments ?

File: src/store/modifier/state.js [hunk 7]
```diff
@@ -1,3 +1,4 @@
 export default {
   recs: [],
+  currentMenuModifier: [],
```
Comment (hemvanh) [inline src/store/modifier/state.js]: currentMenuModifier -> currentMenuModifiers

File: src/components/ModifiersGroup.vue [hunk 8]
```diff
@@ -41,12 +46,44 @@ export default {
       return formatedPrice
     },
   },
+  methods: {
+    ...mapMutations('modifier', ['setCurrentMenuModifier']),
+    upDateModifiers(newModifiers, oldModifiers) {
```
Comment (hemvanh) [inline src/components/ModifiersGroup.vue]: upDateModifiers -> updateModifiers

**Review by hemvanh — CHANGES_REQUESTED** [review]

File: src/components/ModifiersGroup.vue (hunk 2 above)
Comment (hemvanh) [reply src/components/ModifiersGroup.vue]: havent fixed this

File: src/pages/MenuDetail.vue [hunk 9]
```diff
@@ -47,26 +51,44 @@ export default {
...
+    calculateMenuPrice() {
+      this.modifierId = _.map(this.getCurrentMenuModifiers, 'id').sort()
+      let menuPrice =
+        parseFloat(this.menu.price) +
+        _.sumBy(this.getCurrentMenuModifiers, ({price}) => {
+          return parseFloat(price)
+        })
+      this.totalMenuPrice = this.counter * menuPrice
+    },
+    addToCard() {
```
Comment (hemvanh) [inline src/pages/MenuDetail.vue]: -> addToCart

**Review by hemvanh — CHANGES_REQUESTED** [review]

File: src/components/ModifiersGroup.vue [hunk 10]
```diff
@@ -15,6 +15,18 @@
...
+  let currentModifiers = this.getCurrentMenuModifiers
+  if (oldModifiers) {
+    let oldModifierIds = []
+    if (_d.isArray(oldModifiers)) oldModifierIds = _d.map(oldModifiers, 'id')
+    else oldModifierIds = [oldModifiers.id]
+    _d.remove(currentModifiers, ({id}) => {
+      return oldModifierIds.includes(id)
+    })
+  }
+  this.setCurrentMenuModifier(_d.concat(currentModifiers, newModifiers))
```
Comment (hemvanh) [inline src/components/ModifiersGroup.vue]: setCurrentMenuModifier -> setCurrentMenuModifiers

File: src/pages/MenuDetail.vue [hunk 11]
```diff
@@ -47,26 +51,44 @@ export default {
...
+        parseFloat(this.menu.price) +
+        _.sumBy(this.getCurrentMenuModifiers, ({price}) => {
+          return parseFloat(price)
+        })
+      this.totalMenuPrice = this.counter * menuPrice
+    },
+    addToCart() {
+      this.setRecsModifier({
+        menuId: this.menu.id,
+        modifierId: this.modifierId,
```
Comment (hemvanh) [inline src/pages/MenuDetail.vue]: modifierIds ??

File: src/pages/MenuDetail.vue [hunk 12]
```diff
@@ -47,26 +51,44 @@ export default {
...
     ...mapGetters('menu', ['getRecs', 'getCounter']),
     ...mapGetters('modifier', {getRecsModifier: 'getRecs'}),
+    ...mapGetters('modifier', ['getCurrentMenuModifiers']),
   },
   methods: {
     ...mapActions('modifier', ['fetchModifiers']),
     ...mapMutations('menu', ['setCounter']),
-    ...mapMutations('modifier', ['setRecs']),
+    ...mapMutations('modifier', ['setRecs', 'setCurrentMenuModifier']),
+    ...mapMutations('customerorder', {setRecsModifier: 'setRecs'}),
```
Comment (hemvanh) [inline src/pages/MenuDetail.vue]: ...mapMutations('customerorder', {setRecsModifier: 'setRecs'}),

-> ...mapMutations('customerorder', {addMenuToOrder: 'setRecs'}),

**Review by hemvanh — CHANGES_REQUESTED** [review]

File: src/pages/MenuDetail.vue (hunk 11 above)
Comment (tri-et) [reply src/pages/MenuDetail.vue]: u mean modifierId->modifierIds or what's purpose using 'modifierId'?
modifierIds contain those id modifiers that menu chosen

File: src/components/ModifiersGroup.vue [hunk 13]
```diff
@@ -26,8 +27,12 @@ export default {
   data() {
     return {
       defaultModifiers: [],
+      oldModifiers: [],
```
Comment (hemvanh) [inline src/components/ModifiersGroup.vue]: oldModifiers --> no use right ?

**Review by hemvanh — CHANGES_REQUESTED** [review]

**Pull request merged**